homeassistant/helpers/device_registry.py

```diff
@@ -503,14 +503,16 @@ async def _async_migrate_func(
                 # Introduced in 2024.7
                 for device in old_data["devices"]:
                     device["primary_config_entry"] = None
-            if old_minor_version < 7:
+            if old_minor_version < 8:
                 # Introduced in 2024.8
                 created_at = utc_from_timestamp(0).isoformat()
                 for device in old_data["devices"]:
-                    device["model_id"] = None
-                    device["created_at"] = device["modified_at"] = created_at
+                    device.setdefault("model_id", None)
+                    device.setdefault("created_at", created_at)
+                    device.setdefault("modified_at", created_at)
```

Comment:
Not sure the `setdefault` is motivated, we don't need to care about users of nightly having the model_id and timestamps reset.

Reply:
removed